lib/date_time/date_time_ntp.c

```diff
@@ -108,7 +109,9 @@ int date_time_ntp_get(int64_t *date_time_ms)
 			continue;
 		}
 		LOG_DBG("Time obtained from NTP server %s", servers[i]);
-		*date_time_ms = (int64_t)sntp_time.seconds * 1000;
+
+		milliseconds = (uint32_t)(((double) sntp_time.fraction / UINT32_MAX) * 1000);
```

Comment:
I think a comment would be nice to say that the fraction part is not milliseconds, microseconds or nanoseconds but rather fractions of a second based on 32 bit unsigned integer maximum value.
I know that's what the code does or "says" but given this is very weird unit, it would be good to have a comment alarming the reader about it.